trio/_tests/check_type_completeness.py

```diff
@@ -0,0 +1,138 @@
+    last_dict: dict[str, Any] = {}
+
+    def check_less_than(key, /, invert=False):
+        global failed
+        current = current_dict[key]
+        last = last_dict[key]
+        assert isinstance(current, (float, int))
+        assert isinstance(last, (float, int))
+        if current == last:
+            return
+        if (current > last) ^ invert:
+            failed = True
+            print("ERROR: ", end="")
+        if isinstance(current, float):
+            strcurrent = f"{current:.4}"
+            strlast = f"{last:.4}"
+        else:
+            strcurrent = str(current)
+            strlast = str(last)
+        print(
+            f"{key} has gone {'down' if current<last else 'up'} from {strlast} to {strcurrent}"
+        )
+
+    def check_zero(key):
+        if current_dict[key] != 0:
+            global failed
+            failed = True
+            print(f"ERROR: {key} is {current_dict[key]}")
+
+    res = run_pyright()
+    current_result = json.loads(res.stdout)
+
+    # check if py.typed file was missing
+    if (
+        current_result["generalDiagnostics"]
+        and current_result["generalDiagnostics"][0]["message"]
+        == "No py.typed file found"
+    ):
+        print("creating py.typed")
+        py_typed_file = (
+            Path(current_result["typeCompleteness"]["packageRootDirectory"])
+            / "py.typed"
+        )
+        py_typed_file.write_text("")
+
+        res = run_pyright()
+        current_result = json.loads(res.stdout)
+
+    if res.stderr:
+        print(res.stderr)
+
+    last_result = json.loads(RES_FILE.read_text())
+    current_dict = current_result["summary"]
+
+    for key in "errorCount", "warningCount", "informationCount":
+        check_zero(key)
+
+    current_dict = current_result["typeCompleteness"]
+    last_dict = last_result["typeCompleteness"]
+    for key in (
+        "missingFunctionDocStringCount",
+        "missingClassDocStringCount",
+        "missingDefaultParamCount",
+    ):
+        check_less_than(key)
+    check_less_than("completenessScore", invert=True)
+
+    current_dict = current_dict["exportedSymbolCounts"]
+    last_dict = last_dict["exportedSymbolCounts"]
+
+    for key in "withUnknownType", "withAmbiguousType":
+        check_less_than(key)
+    check_less_than("withKnownType", invert=True)
+
+    assert (
+        res.returncode != 0
+    ), "Fully type complete! Replace this script with running pyright --verifytypes directly in CI and checking exit code."
+
+    if args.overwrite_file:
+        print("Overwriting file")
+
+        # don't care about differences in time taken
+        del current_result["time"]
+        del current_result["summary"]["timeInSec"]
+
+        # don't fail on version diff so pyright updates can be automerged
+        del current_result["version"]
+
+        for key in (
+            # don't save huge file for now
+            "symbols",
+            "modules",
+            # don't save path
```

Comment:
I know why this is here but it might be nice to include a short "because that varies from machine to machine" to lessen thinking about this later.